From 3b1526aef691097b61c993e8844a4eff8ffc141a Mon Sep 17 00:00:00 2001
From: Thomas Bouvier <contact@thomas-bouvier.io>
Date: Tue, 17 Oct 2023 12:11:34 +0200
Subject: [PATCH 1/2] feat: compile using cpp17 from version pytorch 2.1

Signed-off-by: Thomas Bouvier <contact@thomas-bouvier.io>
---
 horovod/torch/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/horovod/torch/CMakeLists.txt b/horovod/torch/CMakeLists.txt
index 7234d741ef..d54329d9ff 100644
--- a/horovod/torch/CMakeLists.txt
+++ b/horovod/torch/CMakeLists.txt
@@ -63,7 +63,9 @@ list(APPEND PYTORCH_LINKER_LIBS ${Pytorch_LIBRARIES})
 parse_version(${Pytorch_VERSION} VERSION_DEC)
 add_definitions(-DPYTORCH_VERSION=${VERSION_DEC} -DTORCH_API_INCLUDE_EXTENSION_H=1)
 set(Pytorch_CXX11 ${Pytorch_CXX11} PARENT_SCOPE)
-if(NOT Pytorch_VERSION VERSION_LESS "1.5.0")
+if(NOT Pytorch_VERSION VERSION_LESS "2.1.0")
+    set(CMAKE_CXX_STANDARD 17)
+elseif(NOT Pytorch_VERSION VERSION_LESS "1.5.0")
     set(CMAKE_CXX_STANDARD 14)
 endif()
 
